docs/src/operate/troubleshooting/tedge-diag.md

```diff
@@ -0,0 +1,137 @@
+It is invoked by `tedge diag collect` (referred to as the _runner_).
+
+Each plugin runs as a child process, and its `stdout` and `stderr` are automatically captured by the runner.
+Furthermore, each plugin is executed with command-line arguments that include its dedicated output directory path, where it can create any files.
+These files are then bundled into the final tarball archive.
+
+Plugins are intended to be customized according to your system and diagnostic needs.
+
+### Plugin locations
+
+If %%te%% is installed via a [package](../../install/index.md#installupdate),
+pre-defined diagnostic plugins are installed in `/usr/share/tedge/diag-plugins`.
+
+In addition, multiple plugin directories are supported.
+You can specify multiple `--plugin-path` options:
+
+```sh
+tedge diag collect --plugin-path /usr/share/tedge/diag-plugins --plugin-path /your/own/dir/path
+```
+
+Alternatively, you can add a custom directory path permanently by `tedge config add` command.
+In this case, you don't need to give specific path by command-line.
+
+```sh
+sudo tedge config add diag.plugin_paths "your/own/dir/path"
+tedge diag collect
+```
+
+### Disabling plugins
+
+To diable a plugin, add `.ignore` extension to its filename (e.g., `10_test.sh.ignore`).
+
+### Predefined plugins
+
+The [predefined plugins](https://github.com/thin-edge/thin-edge.io/tree/main/configuration/contrib/diag-plugins) include:
+
+| Plugin Name        | Description                                                      |
+| ------------------ | ---------------------------------------------------------------- |
+| 01_tedge.sh        | Collects logs from `tedge-mapper` and `tedge-agent`              |
+| 02_os.sh           | Collects system information                                      |
+| 03_mqtt.sh         | Collects output of `tedge mqtt sub`                              |
+| 04_workflow.sh     | Copies workflow logs                                             |
+| 05_entities.sh     | Collects entity data                                             |
+| 06_internal.sh     | Copies internal backup files                                     |
+| 07_mosquitto.sh    | Collects mosquitto logs (skipped when using the built-in bridge) |
+| template.sh.ignore | A template for creating custom diagnostic plugin                 |
+
+
+### Writing your own diagnostic plugin
+
+The easiest way to create a custom plugin is to start from the [template script](https://github.com/thin-edge/thin-edge.io/blob/main/configuration/contrib/diag-plugins/template.sh.ignore).
+
+First, rename it to remove the `.ignore` extension and copy it to your desired location:
+
+```sh
+mkdir -p /etc/tedge/diag-plugins
+cp /usr/share/tedge/diag-plugins/template.sh.ignore /etc/tedge/diag-plugins/100_my-plugin.sh
+```
+
+Open the file in a text editor and modify the `collect()` function.
```

Comment:
I would not state the obvious as "Open the file in a text editor" but the goal.

```suggestion
Then modify the `collect()` function to gather all relevant diagnostic data for your service, by printing status data and collecting log files.
```

Reply:
Applied in 8b134ec29235ed4d94ffe953e0b1ad2c0685221b